src/bin/cargo/cli.rs

```diff
@@ -473,13 +475,17 @@ See 'cargo help <command>' for more information on a specific command.\n",
             .global(true),
         )
         .arg(
-            Arg::with_name("unstable-features")
+            Arg::new("unstable-features")
                 .help("Unstable (nightly-only) flags to Cargo, see 'cargo -Z help' for details")
-                .short("Z")
+                .short('Z')
                 .value_name("FLAG")
-                .multiple(true)
-                .number_of_values(1)
+                .multiple_occurrences(true)
                 .global(true),
         )
         .subcommands(commands::builtin())
 }
+
+#[test]
+fn verify_cli() {
+    cli().debug_assert();
+}
```

Comment:
Clap assumes errors are programmer mistakes and reports them through debug_asserrts.  It does this lazily, only walking evaluating the subcommands that the user activates.  This will eagerly run all of the debug asserts, helping to catch problems sooner.